### PR TITLE
Typing fixes

### PR DESCRIPTION
**Pull request by blueyed**
`mypy starlette tests --ignore-missing-imports --check-untyped-defs` passed (except for #780).

### CODE REVIEW

File: starlette/endpoints.py [hunk 1]
```diff
@@ -10,6 +10,14 @@
 from starlette.types import Message, Receive, Scope, Send
 from starlette.websockets import WebSocket
 
+if False:  # TYPE_CHECKING
```
Comment (florimondmanca) [inline starlette/endpoints.py:13]: There's a built-in flag in `typing` for this :-)
```suggestion
if typing.TYPE_CHECKING:
```

Comment (blueyed) [reply starlette/endpoints.py:13]: Yes, makes sense (given that this is py36+), thanks!
Not going to update it for now though, given its staleness already.

**Comment by and-semakin** [comment]
What's the status of this PR? It contains the fix of one little typing annoyance I was going to suggest.

File: starlette/testclient.py [hunk 2]
```diff
@@ -449,7 +449,7 @@ def websocket_connect(
 
         return session
 
-    def __enter__(self) -> requests.Session:
+    def __enter__(self) -> "TestClient":
```
Comment (and-semakin) [inline starlette/testclient.py:452]: Would be nice to have this fix merged.

Comment (and-semakin) [reply starlette/testclient.py:452]: Here we go #1064

**Pull request closed without merging**